### PR TITLE
Fix wrong val passed to done when name taken, & a bug in npm-name mock

### PR DESCRIPTION
**Pull request by Oblongmana**
This is embarassing - in #125 / #126 I said I was passing `true` to `done(...)` in fact I had set it to false in my fork, but to `true` in the local copy I was using for testing - so in what got merged, we'll continue correctly if the name is not taken in npm, but if the name is taken, it doesn't re-prompt. Corrected that in this PR. Triple-checked I haven't made any silly mistakes again this time :D

### CODE REVIEW

**Comment by Oblongmana** [comment]
Hrm, broke the tests again, digging into them now

**Comment by Oblongmana** [comment]
Worked it out, new commit incoming!

**Comment by Oblongmana** [comment]
Discovered there was also a bug with npm-name mock - only providing one param to callback (err), when should have been providing (err, available)

File: generators/app/index.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ module.exports = generators.Base.extend({
             if (available) {
               done();
             } else {
-              done(false);
+              done(true); //re-prompt
```
Comment (arthurvr) [inline generators/app/index.js]: This comment is kinda moot imho.

File: test/app.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ describe('node:app', function () {
     mockery.enable({warnOnUnregistered: false});
 
     mockery.registerMock('npm-name', function (name, cb) {
-      cb(true);
+      cb(null,true);//no errors, name available
```
Comment (arthurvr) [inline test/app.js]: Same thing here. The comment is kinda moot.

Comment (Oblongmana) [inline test/app.js]: For sure, had them there while debugging, but certainly not needed now. Will fix those up

File: test/app.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ describe('node:app', function () {
     mockery.enable({warnOnUnregistered: false});
 
     mockery.registerMock('npm-name', function (name, cb) {
-      cb(true);
+      cb(null,true);
```
Comment (arthurvr) [inline test/app.js]: Needs a space between those two arguments `null, true`

**Comment by arthurvr** [comment]
Thanks!

**Comment by Oblongmana** [comment]
Happy to help! :D

**Pull request closed without merging**